chain/default_syncer.go

```diff
@@ -190,10 +189,19 @@ func (syncer *DefaultSyncer) collectChain(ctx context.Context, tipsetCids types.
 			return nil, err
 		}
 
-		ts, err := syncer.consensus.NewValidTipSet(ctx, blks)
+		// validate blocks we maybe got from the network
+		// TODO: if we didn't get these blocks from the network and instead got
+		// them from disk, validation feels redundant. This requires
+		// looser coupling between the store and the fetcher.
```

Comment:
Agreed, this should happen further out. If this is the pragmatic place to put it right now, then ok.

When pushed out to the network, a syntactically invalid block won't even make it as far as badTipSets.

Don't add a TODO without an issue, but here are two prepared earlier: #2128, #2783

Reply:
Awesome, thank you for that link -- will move this validation check to the fetcher.